src/options.py

```diff
@@ -107,7 +107,7 @@ class GlobalDeclsEnum(ChoicesEnum):
     use_cache: bool
     dump_typemap: bool
     pdb_translate: bool
-    preproc_defines: Dict[str, int]
+    preproc_defines: Dict[str, Optional[int]]
```

Comment:
Maybe a comment here or somewhere that `None` represents unset macros, and `1` is used for macros set without a value? 
Naively I might expect that `-D FOO` would set `preproc_defines["FOO"] = None` instead of `1`.